sshd-common/src/main/java/org/apache/sshd/common/util/buffer/Buffer.java

```diff
@@ -341,6 +341,39 @@ public String getString() {
         return GenericUtils.isEmpty(values) ? Collections.emptyList() : Arrays.asList(values);
     }
 
+    public List<OpenSshCertificate.CertificateOption> getCertificateOptions() {
+        return getCertificateOptions(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * According to <A HREF=
+     * "https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.certkeys#L222-L262">PROTOCOL.certkeys</A>:
+     *
+     * Critical Options is a set of bytes that is
+     *
+     * [overall length][name(string)][data(string)]...
+     *
+     * Where each Critical Option is encoded as a name (string) and data (string)
+     *
+     * Then the entire name + data strings are added as bytes (which will get a length prefix)
+     *
+     * @return The parsed result
+     */
+    public List<OpenSshCertificate.CertificateOption> getCertificateOptions(Charset charset) {
+        // pull out entire Critical Options section
+        final ByteArrayBuffer criticalOptionsBuf = new ByteArrayBuffer(getBytes());
+
+        List<OpenSshCertificate.CertificateOption> list = new ArrayList<>();
+
+        while (criticalOptionsBuf.available() > 0) {
+            String name = criticalOptionsBuf.getString(charset);
+            String data = GenericUtils.trimToEmpty(criticalOptionsBuf.getString(charset));
+            list.add(new OpenSshCertificate.CertificateOption(name, data.length() > 0 ? data : null));
+        }
+
+        return list.size() > 0 ? list : null;
```

Comment:
In general i think returning null for collections is bad, but I see that here it may make sense given how these lists are passed around through getters and setters. Fixing that would have to start already at OpenSshCertificateImpl.getCertificateOptions().

So OK, let's leave it with returning null. I'd use list.isEmpty(), though.

Please add javadoc to OpenSshCertificate documenting that it may return null or an empty list.